programs/mango-v3-reimbursement/src/instructions/reimburse.rs

```diff
@@ -0,0 +1,129 @@
+use anchor_lang::{__private::bytemuck, prelude::*};
+use anchor_spl::token::{self, Mint, Token, TokenAccount};
+
+use crate::state::{Group, ReimbursementAccount, Table};
+
+#[derive(Accounts)]
+#[instruction(token_index: usize)]
+pub struct Reimburse<'info> {
+    #[account (
+        constraint = group.load()?.has_reimbursement_started()
+    )]
+    pub group: AccountLoader<'info, Group>,
+
+    #[account(
+        mut,
+        constraint = group.load()?.vaults[token_index] == vault.key()
+    )]
+    pub vault: Account<'info, TokenAccount>,
+
+    #[account(
+        constraint = group.load()?.mints[token_index] == mint.key()
+    )]
+    pub mint: Box<Account<'info, Mint>>,
+
+    #[account(
+        mut,
+        constraint = token_account.owner == mango_account_owner.key()
+    )]
+    pub token_account: Box<Account<'info, TokenAccount>>,
+    #[account(
+        mut,
+        // TODO: enable after testing is done
+        // constraint = !reimbursement_account.load()?.reimbursed(token_index),
+        // constraint = !reimbursement_account.load()?.claim_transferred(token_index),        
+    )]
+    pub reimbursement_account: AccountLoader<'info, ReimbursementAccount>,
```

Comment:
Why do you need to check the address manually later? A `seeds` constraint here should work?